packages/eip-1193-provider/src/errors.ts

```diff
@@ -0,0 +1,114 @@
+import { ProviderRpcError } from 'viem';
+import { PROVIDER_ERROR_CODE, PROVIDER_ERROR_MESSAGE } from './constants';
+
+class UnrecognizedChainError extends ProviderRpcError {
+  constructor(chainId: string) {
+    const errorMessage = `${PROVIDER_ERROR_MESSAGE.UNRECOGNIZED_CHAIN_ID} Received chainId: ${chainId}`;
```

Comment:
Nit: fold the "Received: " into the `UNRECOGNIZED_CHAIN_ID` message like you've done with the other error messages?